lib/engine/tile.rb

```diff
@@ -395,6 +395,14 @@ def city_town_edges
       ct_edges.values
     end
 
+    def city_town_edges_are_subset_of?(other_cte)
+      cte = city_town_edges
+      ALL_EDGES.any? do |rotation|
```

Comment:
can this just be done inline like

ALL_EDGES.any? do |rotation|
  other_cte.any? do | city| 
       city.any ? | edge|
        ( rotate(edge, rotation) - other).empty?

Reply:
I was trying to avoid calling `city_town_edges` repeatedly since this will have to recalculate from the cities on the tile, so is the expensive part of the calculation. Therefore I call this once and store in `cte`, and then rotate-and-compare it to the data from the other tile which has been passed in as an argument to this method.

Reply:
We need to make sure all that all edges of all cities are present on the new tile, so I guess it'd be:
```
cte = city_town_edges
ALL_EDGES.any? do |rotation|
    cte.all? do |city|
        other_cte.any? do |other_city|
            city.all? |edge| other_cte.include?(rotate(edge, rotation))
        end
    end
end
```

Is that better? It feels less readable; does the `.map`have a significant overhead we should be avoiding?

Reply:
cool